File: rb/features.rb
# RestApiBeispiele SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RestApiBeispieleFeatures
  def self.make_feature(name)
    case name
    when "base"
      RestApiBeispieleBaseFeature.new
    when "test"
      RestApiBeispieleTestFeature.new
    else
      RestApiBeispieleBaseFeature.new
    end
  end
end

File: py/restapibeispiele_sdk/features.py
# RestApiBeispiele SDK feature factory

from restapibeispiele_sdk.feature.base_feature import RestApiBeispieleBaseFeature
from restapibeispiele_sdk.feature.test_feature import RestApiBeispieleTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RestApiBeispieleBaseFeature(),
        "test": lambda: RestApiBeispieleTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
